carbon-design-system/carbon-addons-iot-react
# feat(progress-bar): initial addition of new progress bar component

Closes #3215 

**Summary**

- Adds the new ProgressBar component--which I'm not sure is a good name for this component--the ability to change the colors based on threshold levels seems lend more to a StatusBar component (perhaps?). It is built on the experimental Carbon ProgressBar, adds the text value label and icon (when needed) to the top-right side of the component, and gives the ability to pass an array of thresholds. These thresholds match what is already in place with ValueCards with a couple minor differences. The comparison can be a simple `>`, `>=`, `=`, `<`, `<=`, or it can be a function. The addition of the function allows for more complex comparisons like "greater-than-or-equal-to X AND less-than-Y". Also, the color can be provided as a simple string hex color, or as an object with `fill` and `stroke` keys to meet the requirements of certain colors requiring strokes to make them a11y compliant. 

**Change List (commits, features, bugs, etc)**

- added useMerged hook from v3
- added new `useMatchedThreshold` hook and refactored in ValueCard
- added new ProgressBar component and styles
- added tests
- exported from index

**Acceptance Test (how to verify the PR)**

- check the new [ProgressBar stories](https://deploy-preview-3232--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-progressbar--default) and compare to the specs in the original issue
- repeat in RTL
- check the [ProgressBar with thresholds](https://deploy-preview-3232--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-progressbar--with-thresholds) stories
- confirm the colors change with each threshold value `< 80`, `>=80 AND <90`, `>=90 AND <95`, `>= 95 AND <= 100`, `>100` (should also turn value text red and show icon`
- run a11y checks at various thresholds to confirm no issues
- proofread documentation and check for missing props 
- confirm tests cover necessary requirements

**Regression Test (how to make sure this PR doesn't break old functionality)**

- n/a



**Things to look for during review**
